llvm/unittests/ADT/APFloatTest.cpp

```diff
@@ -6532,6 +6534,34 @@ TEST(APFloatTest, Float8E5M2ToDouble) {
   EXPECT_TRUE(std::isnan(QNaN.convertToDouble()));
 }
 
+TEST(APFloatTest, Float8E4M3ToDouble) {
+  APFloat One(APFloat::Float8E4M3(), "1.0");
+  EXPECT_EQ(1.0, One.convertToDouble());
+  APFloat Two(APFloat::Float8E4M3(), "2.0");
+  EXPECT_EQ(2.0, Two.convertToDouble());
+  APFloat PosLargest = APFloat::getLargest(APFloat::Float8E4M3(), false);
+  EXPECT_EQ(240.0F, PosLargest.convertToDouble());
+  APFloat NegLargest = APFloat::getLargest(APFloat::Float8E4M3(), true);
+  EXPECT_EQ(-240.0F, NegLargest.convertToDouble());
+  APFloat PosSmallest =
+      APFloat::getSmallestNormalized(APFloat::Float8E4M3(), false);
+  EXPECT_EQ(0x1.p-6, PosSmallest.convertToDouble());
+  APFloat NegSmallest =
+      APFloat::getSmallestNormalized(APFloat::Float8E4M3(), true);
+  EXPECT_EQ(-0x1.p-6, NegSmallest.convertToDouble());
```

Comment:
Sanity check: this test is basically exactly the same as above except with exponent adjusted?

Reply:
llvm/clang/APFloat code is from [PR-97179](https://github.com/llvm/llvm-project/pull/97179).

Reply:
f8E4M3 Min normal number: S.0001.000 = +/-2^(-6)
```
APFloat::getSmallestNormalized(APFloat::Float8E4M3(), false) // returns 0.0001.000 => +2^(-6)
APFloat::getSmallestNormalized(APFloat::Float8E4M3(), true) // returns 1.0001.000 => -2^(-6)
```